lib/API/auth.dart

```diff
@@ -0,0 +1,67 @@
+import 'dart:convert';
+import 'package:app/model/user.dart';
+import 'package:http/http.dart' as http;
+
+class Auth {
+  static String accessToken = "";
+  static int userID = 0;
+  static User user = User();
+  static List<User> allUsers = [];
+
+  // Function to login
+  static Future<bool> login(var basicAuth) async {
+    var url = Uri.parse('http://161.35.99.225/api/v1/auth/login');
+    var response = await http
+        .post(url, headers: <String, String>{'authorization': basicAuth});
+    if (response.statusCode == 200) {
+      var jsonResponse = jsonDecode(response.body);
+      jsonResponse = jsonResponse['result'];
+      Auth.accessToken = jsonResponse['accessToken'];
+      Auth.userID = jsonResponse['id'];
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  // function to store users assuming that the user is logged in
+  static Future storeUser() async {
+    var url = Uri.parse('http://161.35.99.225/api/v1/users/$userID');
+    var response = await http.get(
+      url,
+      headers: <String, String>{'authorization': accessToken},
+    );
+    if (response.statusCode == 200) {
+      var jsonResponse = jsonDecode(response.body);
+      jsonResponse = jsonResponse['result'];
+      user = User.fromJson(jsonResponse);
+      return true;
+    } else {
+      return false;
+    }
+  }
```

Comment:
The name of this function is `storeUser`. But it actually fetches the user and then stores it. 
You should breakdown the login into two functions. One for fetching the user, the second is for storing the user.